tests/test_checkout_integration.py

```diff
@@ -306,3 +357,427 @@ def test_remove_voucher(client, request_cart_with_item, shipping_method, voucher
                                    follow=True, HTTP_REFERER=url)
     assert voucher_response.status_code == 200
     assert voucher_response.context['checkout'].voucher_code is None
+
+
+def test_user_pass_new_valid_shipping_address(authorized_client, customer_user,  # pylint: disable=W0613,R0914
+                                              request_cart_with_item, shipping_method):  # pylint: disable=W0613
+    """User pass new valid shipping address
+      - if is save in session storage
+      - if is save in order object, after finish checkout
+      - if is add to authorized user (ass default shipping address)
+      user pass new valid shipping address, and use it as billing address
+      - if is save in order
+      - if is save in as default
+
+    """
+
+    # Enter checkout
+    checkout_index = authorized_client.get(reverse('checkout:index'), follow=True)
+    # Checkout index redirects directly to shipping address step
+    shipping_address = authorized_client.get(checkout_index.request['PATH_INFO'])
+
+    # Enter shipping address data
+    shipping_data = {
```

Comment:
What's wrong with having a `valid_address` fixture? The address is not the point of this test.